ls1intum/Artemis
# `Development`: Avoid unnecessary REST calls for practice participation by loading them in the for-dashboard call

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Improve the performance of the exercise overview

### Description

Each time a row was extended in the exercise overview, the exercise details were loaded again to find the participations of the exercises. The loading of the participations is now taken care of in the for-dashboard call and removed, so it only has to be loaded once.
Since the practice participations are only needed in the specific courses, they are not included in the general for-dashboard call.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise
- 1 Modeling Exercise
- 1 Quiz Exercise

1. Log in to Artemis
2. Navigate to the exercise overview and check that all exercises, their scores and their action buttons are displayed correctly.
3. Check that there is no "details" call when expanding a row in the exercise overview
4. Wait after the due date of the programming exercise (or set it now to the past)
5. Use the practice mode in the programming exercise and submit a new partial correct solution
6. Check that the correct student actions (only cloning/online editor) are shown in the exercise overview even when reloading the page
7. Check that the score in the exercise row still shows the graded score

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ProgrammingExercise.java | 92% | ✅                           |
